libloki/crypto.js

```diff
@@ -324,6 +324,144 @@
     GRANT: 2,
   });
 
+  /**
+   * A wrapper around Signal's SessionCipher.
+   * This handles specific session reset logic that we need.
+   */
+  class LokiSessionCipher {
+    constructor(storage, address) {
+      this.storage = storage;
+      this.address = address;
```

Comment:
any reason not to just toString() it here?

Reply:
because this address is actually `SignalProtocolAddress`. I'll change the name of it in a commit to make it more clear.

`address.toString()` gives us the public hex encoded key concatenated with the device id.
`address.getName()` gives us just the public hex encoded key.

We require both of them so i left is as is.

Reply:
ah ok

Reply:
There's not much cpu overhead as it's basically:
```js
  toString: function() {
    return this.name + '.' + this.deviceId;
  },
```